cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -251,266 +229,35 @@ While all the techniques referenced here do not require any user interaction, so
-            -> c.getName().equals(csrfCookieExpectedName)).findFirst().orElse(null);
-        }
-        if (tokenCookie == null || this.isBlank(tokenCookie.getValue())) {
-            LOG.info("CSRFValidationFilter: CSRF cookie absent or value" + 
-            " is null/empty so we provide one and return an HTTP NO_CONTENT response !");
-            //Add the CSRF token cookie and header
-            this.addTokenCookieAndHeader(httpReq, httpResp);
-            //Set response state to "204 No Content" in order to allow the requester to 
-            //clearly identify an initial response providing the initial CSRF token
-            httpResp.setStatus(HttpServletResponse.SC_NO_CONTENT);
-        } else {
-            //If the cookie is present then we pass to validation phase
-            //Get token from the custom HTTP header (part under control of the requester)
-            String tokenFromHeader = httpReq.getHeader(CSRF_TOKEN_NAME);
-            //If empty then we trace the event and we block the request
-            if (this.isBlank(tokenFromHeader)) {
-                accessDeniedReason = "CSRFValidationFilter: Token provided via HTTP Header"+ 
-                " is absent/empty so we block the request !";
-                LOG.warn(accessDeniedReason);
-                httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-            } else if (!tokenFromHeader.equals(tokenCookie.getValue())) {
-                //Verify that token from header and one from cookie are the same
-                //Here is not the case so we trace the event and we block the request
-                accessDeniedReason = "CSRFValidationFilter: Token provided via HTTP Header"+ 
-                "and via Cookie are not equals so we block the request !";
-                LOG.warn(accessDeniedReason);
-                httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-            } else {
-                //Verify that token from header and one from cookie matches
-                //Here is the case so we let the request reach the target component 
-                //(ServiceServlet, jsp...) and add a new token when we get back the bucket
-                HttpServletResponseWrapper httpRespWrapper = 
-                                            new HttpServletResponseWrapper(httpResp);
-                chain.doFilter(request, httpRespWrapper);
-                //Add the CSRF token cookie and header
-                this.addTokenCookieAndHeader(httpReq, httpRespWrapper);
-            }
-        }
-    }
+Login CSRF can be mitigated by creating pre-sessions (sessions before a user is authenticated) and including tokens in login form. You can use any of the techniques mentioned above to generate tokens. Pre-sessions can be transitioned to real sessions once the user is authenticated. This technique is described in [Robust Defenses for Cross-Site Request Forgery section 4.1](https://seclab.stanford.edu/websec/csrf/csrf.pdf).
```

Comment:
Sessions should never be transitions 9you end up with session fixation) - the session should be destroyed and a new one made.

You could also put a CAPTCHA/2FA on your login form.